File: VIF_git/GSDS-Qual/LeetCode/week2/Backtracking/Combination_Sum_III/suho/CS.cpp
class Solution {
public:
    void combi(int k, int n, vector<int> l, vector<vector<int>> &agg){
        if(l.size()>=k){
            int sum = 0;
            for(int i = 0; i< k; i++) sum+=l[i];
            if(sum==n) agg.push_back(l);
            return;
        }
        else if (l.back()==9) return;
        else{
            for(int i = l.back()+1;i<10;i++){
                vector<int> l2;
                l2.resize((int)(l.size()));
                copy( l.begin(), l.end(), l2.begin() );
                l2.push_back(i);
                combi(k, n, l2, agg);
            }
            return;
        }
    }

    vector<vector<int>> combinationSum3(int k, int n) {
        vector<vector<int>> agg(0);
        for(int i = 1; i<10; i++){
            combi(k, n, vector<int>{i}, agg);
        }
            
        return agg;
    }
};

File: VIF_git/GSDS-Qual/LeetCode/week2/Backtracking/Combination_Sum_III/suho/CS.py
class Solution:
    def combi(self, k, n, l, agg):
        if len(l)==k:
            if sum(l)==n: agg.append(l)
            return
        elif l[-1]==9: return
        else: 
            for i in range(l[-1]+1,10):
                l2 = l+[i]
                self.combi(k, n, l2, agg)

    def combinationSum3(self, k, n):
        agg=[]
        for i in range(1,10):
            self.combi(k, n, [i], agg)
        
        return agg

        
